main-script.cpp

```diff
@@ -0,0 +1,79 @@
+#include <iostream>
+#include "matrix-matrix-product.hpp"
+#include "matrix-vector-product.hpp"
+#include "read-matrix-market.hpp"
+#include "write-matrix-market.hpp"
+
+using namespace std;
+typedef vector<vector<double>> Matrix;
+typedef vector<double> Vector;
```

Comment:
Good use of typedef. For "modern" C++ you should use `using Matrix = vector<vector<double>>` instead of typedef as it can be consistently used for types, namespaces, etc. and can be templated if needed. I.e., it can be more consistently applied and it's more semantically clear especially when you try to write it for a function pointer.